cwms-data-api/src/main/java/cwms/cda/data/dao/JooqDao.java

```diff
@@ -524,6 +533,43 @@ private static InvalidItemException buildInvalidUnits(RuntimeException input) {
         return new InvalidItemException(localizedMessage, cause);
     }
 
+    private static ValueTooLongException buildValueTooLongException(RuntimeException input)
+    {
+        Throwable cause = input.getCause();
+        if (input instanceof DataAccessException) {
+            DataAccessException dae = (DataAccessException) input;
+            cause = dae.getCause();
+        }
+
+        String localizedMessage = cause.getLocalizedMessage();
+        if (localizedMessage != null) {
+            String[] parts = localizedMessage.split("\n");
+            if (parts.length > 1) {
+                localizedMessage = parts[0];
+            }
+        }
+        if (hasCodeOrMessage((SQLException) input.getCause(), Collections.singletonList(12899),
+                Collections.singletonList("value too large for column"))) {
+            if (!localizedMessage.startsWith("ORA-12899:")) {
```

Comment:
IllegalArgumentException should be when a user supplies an illegal argument, not when the database does. If the message format doesn't match what you are expecting, then throw the other `ValueTooLongException` message.

Additionally, use a regex so you can do all the checks at once